AEPCore/Sources/rules/LaunchRulesEngine.swift

```diff
@@ -128,12 +132,27 @@ public class LaunchRulesEngine {
                     eventData = modifiedEventData
 
                 case LaunchRulesEngine.CONSEQUENCE_TYPE_DISPATCH:
+
+                    if let unwrappedDispatchCount = dispatchChainCount {
+                        // Keep track of triggering event in edge case of event being dispatched multiple times
+                        // This only applies to events created from this dispatch consequence, hence the check
+                        // for a null 'dispatchChainCount' prior to adding back to dictionary
+                        dispatchChainedEventsCount[event.id] = LaunchRulesEngine.MAX_CHAINED_CONSEQUENCE_COUNT
+
+                        if unwrappedDispatchCount >= LaunchRulesEngine.MAX_CHAINED_CONSEQUENCE_COUNT {
+                            Log.trace(label: LOG_TAG, "(\(self.name)) : Unable to process a DispatchConsequence Event, max dispatch consequences met for this event, \(event)")
```

Comment:
Rewording suggestion: `Unable to process dispatch consequence, max chained dispatch consequences limit of \(MAX_CHAINED_CONSEQUENCE_COUNT) met for this event uuid \(event.id)`